domain/domain.go

```diff
@@ -103,6 +96,16 @@ func (do *Domain) InfoSchema() infoschema.InfoSchema {
 	return do.infoHandle.Get()
 }
 
+// GetSnapshotInfoSchema gets a snapshot information schema.
+func (do *Domain) GetSnapshotInfoSchema(snapshotTS uint64) (infoschema.InfoSchema, error) {
```

Comment:
Why not use `MustReload` directly？ If use this function it needs to set `SchemaValidity` value.